File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.js
var dict = {
  "zero": 0,
  "one": 1,
  "two": 2,
  "three": 3,
  "four": 4,
  "five": 5,
  "six": 6,
  "seven": 7,
  "eight": 8,
  "nine": 9
};

function isNumber(char) {
  var numericPattern = /^[0-9]$/;
  return numericPattern.test(char);
}

function solution(s) {
  var answer = '';
  var idx = 0;

  while (idx < s.length) {
    if (isNumber(s[idx])) {
      answer += s[idx];
      idx += 1;
    } else {
      for (let ed = idx + 1; ed <= s.length; ed++) {
        var temp = s.slice(idx, ed);
        if (dict[temp] !== undefined) {
          answer += dict[temp];
          idx = ed;
          break;
        }
      }
    }
  }

  return parseInt(answer);
}

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
dic = {'zero' : '0', 'one': '1', 'two': '2', 'three': '3', 'four': '4', 'five': '5', 'six': '6', 'seven': '7', 'eight': '8', 'nine': '9'}

def solution(s):
    answer = ''
    idx = 0
    while idx < len(s):
        if s[idx].isdigit():
            answer += s[idx]
            idx += 1
        else:
            for end in range(idx + 1, len(s) + 1):
                tmp = s[idx:end]
                if tmp in dic:
                    answer += dic[tmp]
                    idx = end
                    break
    return int(answer)

result = solution('23four5six7')
print(result)
